pkg/grpc/jwt_authenticator.go

```diff
@@ -0,0 +1,140 @@
+package grpc
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/buildbarn/bb-storage/pkg/clock"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+// JWTKeyConfig represents an object that can return key content. It is used to bridge
+// configuration to this package.
+type JWTKeyConfig struct {
+	Key interface{}
+}
+
+type jwtAuthenticator struct {
+	verifyKey JWTKeyConfig
+	clock     clock.Clock
+}
+
+// From: https://github.com/square/go-jose/blob/v2/jose-util/utils.go
+func loadJSONWebKey(json []byte, pub bool) (*jose.JSONWebKey, error) {
+	var jwk jose.JSONWebKey
+	err := jwk.UnmarshalJSON(json)
+	if err != nil {
+		return nil, err
+	}
+	if !jwk.Valid() {
```

Comment:
Now that you've locally patched this function to do the right thing, these two functions can be removed... right?